src/embedded.ts

```diff
@@ -215,11 +218,20 @@ export class RoomWidgetClient extends MatrixClient {
 
     private onEvent = async (ev: CustomEvent<ISendEventToWidgetActionRequest>) => {
         ev.preventDefault();
-        const event = new MatrixEvent(ev.detail.data);
-        await this.syncApi.injectRoomEvents(this.room, [], [event]);
-        this.emit(ClientEvent.Event, event);
-        this.setSyncState(SyncState.Syncing);
-        logger.info(`Received event ${event.getId()} ${event.getType()} ${event.getStateKey()}`);
+
+        // Verify the room ID matches, since it's possible for the client to
+        // send us events from other rooms if this widget is always on screen
+        if (ev.detail.data.room_id === this.roomId) {
+            const event = new MatrixEvent(ev.detail.data as Partial<IEvent>);
+            await this.syncApi.injectRoomEvents(this.room, [], [event]);
+            this.emit(ClientEvent.Event, event);
+            this.setSyncState(SyncState.Syncing);
+            logger.info(`Received event ${event.getId()} ${event.getType()} ${event.getStateKey()}`);
+        } else {
+            const { event_id: eventId, room_id: roomId } = ev.detail.data;
+            logger.info(`Received event ${eventId} for a different room ${roomId}; discarding`);
```

Comment:
Nit: I think it might be nicer for this to be in the `if` followed by `return`

Reply:
Returning early would mean that we skip the `ack` call though, which is still something we should do